indico/modules/events/registration/models/invitations.py

```diff
@@ -0,0 +1,137 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2015 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from uuid import uuid4
+
+from sqlalchemy.dialects.postgresql import UUID
+
+from indico.core.db.sqlalchemy import PyIntEnum
+from indico.util.i18n import L_
+from indico.util.locators import locator_property
+from indico.util.string import format_repr, return_ascii
+from indico.util.struct.enum import TitledIntEnum
+from indico.core.db import db
+
+
+class InvitationState(TitledIntEnum):
+    __titles__ = [L_('Pending'), L_('Accepted'), L_('Declined')]
+    pending = 0
+    accepted = 1
+    declined = 2
+
+
+class RegistrationInvitation(db.Model):
+    """An invitation for someone to register"""
```

Comment:
I was even going to suggest calling this `RegistrationFormInvitation` to make it really clear...